source/lib/cli-run-options.test.ts

```diff
@@ -0,0 +1,136 @@
+import { InvalidArgumentError } from 'commander';
+import Maybe, { type Just } from 'true-myth/maybe';
+import { createCliRunOptions, type CliRunOptions, type CreateCliRunOptions } from './cli-run-options.js';
+
+const createCliRunOptionsFactory = Factory.define<CreateCliRunOptions>(() => {
+    return {
+        versionNumber: undefined,
+        commandOptions: {},
+        changelogPath: ''
+    };
+});
+
+const testCreateCliRunOptionsMacro = test.macro(
+    (t, optionsOverrides: DeepPartial<CreateCliRunOptions>, expected: Result<CliRunOptions, InvalidArgumentError>) => {
+        const options = createCliRunOptionsFactory.build(optionsOverrides);
+
+        const actual = createCliRunOptions(options);
+
+        t.deepEqual(actual, expected);
+    }
+);
+
+test(
+    'createCliRunOptions() returns an Result Error when "unreleased" command option exists and a version number was provided',
+    testCreateCliRunOptionsMacro,
+    {
+        commandOptions: {
+            unreleased: true
+        },
+        versionNumber: '1.2.3'
+    },
+    Result.err<CliRunOptions, InvalidArgumentError>(
+        new InvalidArgumentError('A version number is not allowed when --unreleased was provided')
+    )
+);
+
+test(
+    'createCliRunOptions() returns an Result Ok when "unreleased" command option exists and no version number was provided',
+    testCreateCliRunOptionsMacro,
+    {
+        commandOptions: {
+            unreleased: true
+        },
+        versionNumber: undefined
+    },
+    Result.ok<CliRunOptions, InvalidArgumentError>({
+        unreleased: true,
+        versionNumber: Maybe.nothing<string>(),
+        sloppy: false,
+        changelogPath: '',
+        stdout: false
+    })
+);
+
+test(
+    'createCliRunOptions() returns an Result Error when "unreleased" command option does not exists and no version number was provided',
+    testCreateCliRunOptionsMacro,
+    {
+        commandOptions: {
+            unreleased: undefined
+        },
+        versionNumber: undefined
+    },
+    Result.err<CliRunOptions, InvalidArgumentError>(new InvalidArgumentError('Version number is missing'))
+);
+
+test(
+    'createCliRunOptions() returns an Result Error when "unreleased" command option is false and no version number was provided',
+    testCreateCliRunOptionsMacro,
+    {
+        commandOptions: {
+            unreleased: false
+        },
+        versionNumber: undefined
+    },
+    Result.err<CliRunOptions, InvalidArgumentError>(new InvalidArgumentError('Version number is missing'))
+);
+
+test(
+    'createCliRunOptions() returns an Result Ok when "unreleased" command option is false and a version number was provided',
+    testCreateCliRunOptionsMacro,
+    {
+        commandOptions: {
+            unreleased: false
+        },
+        versionNumber: '1.2.3'
+    },
+    Result.ok<CliRunOptions, InvalidArgumentError>({
+        unreleased: false,
+        versionNumber: Maybe.just('1.2.3') as Just<string>,
```

Comment:
```suggestion
        versionNumber: Maybe.just<string>('1.2.3'),
```

Reply:
This is unfortunately not possible because I introduced a discriminated union type of `Just<string> | Nothing<string>`. [Maybe.just()](https://true-myth.js.org/functions/maybe.just-1.html) returns unfortunately a `Maybe` instead of a `Just` whereas [Maybe.nothing()](https://true-myth.js.org/functions/maybe.nothing-1.html) returns a correct `Nothing`. I think this is a bug in [true-myth](https://github.com/true-myth/true-myth) which I wanted to look at at some point in the future.